File: frontend/src/types/currentSeason.ts
export interface CurrentSeasonRosterSummary {
  source_url: string;
  season: string;
  source_snapshot_id: number;
  players_seen: number;
  players_created: number;
  identities_created: number;
  player_seasons_created: number;
  player_seasons_updated: number;
  quality_issues_created: number;
}

export interface CurrentSeasonGameRefresh {
  game_id: number;
  title: string;
  source_url: string;
  reasons: string[];
  status: "refreshed" | "failed";
  error: string | null;
}

export interface CurrentSeasonSync {
  run_id: number;
  sport_slug: string;
  season: string;
  status: "succeeded" | "partial";
  correction_lookback: number;
  started_at: string;
  finished_at: string;
  roster: CurrentSeasonRosterSummary;
  schedule_events_seen: number;
  schedule_games_created: number;
  schedule_games_changed: number;
  schedule_games_unchanged: number;
  final_boxscores_seen: number;
  boxscores_selected: number;
  boxscores_refreshed: number;
  boxscores_skipped: number;
  boxscores_failed: number;
  open_identity_issues: number;
  games: CurrentSeasonGameRefresh[];
}

File: backend/app/schemas/current_season.py
"""Schemas for bounded current-season warehouse synchronization."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.roster import RosterImportRead


class CurrentSeasonGameRefreshRead(BaseModel):
    """One selected final game and the reason it was refreshed."""

    model_config = ConfigDict(from_attributes=True)

    game_id: int
    title: str
    source_url: str
    reasons: list[str]
    status: str
    error: str | None = None


class CurrentSeasonSyncRead(BaseModel):
    """Operator-readable result of one bounded WBB season synchronization."""

    model_config = ConfigDict(from_attributes=True)

    run_id: int
    sport_slug: str
    season: str
    status: str
    correction_lookback: int
    started_at: datetime
    finished_at: datetime
    roster: RosterImportRead
    schedule_events_seen: int
    schedule_games_created: int
    schedule_games_changed: int
    schedule_games_unchanged: int
    final_boxscores_seen: int
    boxscores_selected: int
    boxscores_refreshed: int
    boxscores_skipped: int
    boxscores_failed: int
    open_identity_issues: int
    games: list[CurrentSeasonGameRefreshRead]
